src/core/tests/ov_tensor_test.cpp

```diff
@@ -570,6 +570,50 @@ TEST_F(OVTensorTest, makeRangeRoiStringTensor) {
     ASSERT_EQ(roi_tensor.get_element_type(), t.get_element_type());
 }
 
+TEST_F(OVTensorTest, setSmallerShapeOnRoiTensor) {
+    ov::Tensor t{ov::element::i32, {1, 3, 6, 5}};
+    ov::Tensor roi_tensor{t, {0, 0, 1, 2}, {1, 2, 5, 4}};
+    const ov::Shape newShape({1, 1, 3, 2});
+
+    ASSERT_EQ(roi_tensor.get_shape(), ov::Shape({1, 2, 4, 2}));
+
+    roi_tensor.set_shape(newShape);
+    ASSERT_EQ(roi_tensor.get_shape(), newShape);
+}
+
+TEST_F(OVTensorTest, setMaxSizeShapeOnRoiTensor) {
+    ov::Tensor t{ov::element::i32, {1, 3, 6, 5}};
+    ov::Tensor roi_tensor{t, {0, 0, 1, 2}, {1, 2, 5, 5}};
+    const ov::Shape new_shape({1, 2, 4, 3});
+    const ov::Shape roi_capacity({1, 2, 4, 3});
+
+    ASSERT_EQ(roi_tensor.get_shape(), roi_capacity);
+
+    roi_tensor.set_shape(new_shape);
+    ASSERT_EQ(roi_tensor.get_shape(), new_shape);
+
+    roi_tensor.set_shape(roi_capacity);
+    ASSERT_EQ(roi_tensor.get_shape(), roi_capacity);
```

Comment:
Values of `new_shape` and `roi_capacity` are equal. Shouldn't `new_shape` have smaller dimensions? It's not clear what is tested here.

Reply:
Shape should be diffrent will update.